### ISSUE TITLE
Plugin should suppress PropertyNotSetInConstructor on ServiceSubscriberTrait::$container

### ISSUE THREAD

**Issue by StephenClouse** [psalm/psalm-plugin-symfony#359]
Much like AbstractController, ServiceSubscriberTrait (renamed to ServiceMethodsSubscriberTrait in Symfony 7) builds and injects a scoped container to support lazy loading of services. Currently this is flagged by psalm because the container is setter injected:

> ERROR: PropertyNotSetInConstructor - src/DataProvider/DataProvider.php:22:7 - Property App\DataProvider\DataProvider::$container is not defined in constructor of App\DataProvider\DataProvider or in any private or final methods called in the constructor (see https://psalm.dev/074)
class DataProvider implements ServiceSubscriberInterface

We wrote this tiny hack plugin to suppress it across the board but you probably know a better way:

```php
<?php

declare(strict_types=1);

namespace App\Psalm;

use Psalm\Issue\PropertyNotSetInConstructor;
use Psalm\Plugin\EventHandler\BeforeAddIssueInterface;
use Psalm\Plugin\EventHandler\Event\BeforeAddIssueEvent;

class ServiceSubscriberPlugin implements BeforeAddIssueInterface
{
    public static function beforeAddIssue(BeforeAddIssueEvent $event): ?bool
    {
        $codeIssue = $event->getIssue();

        if ($codeIssue instanceof PropertyNotSetInConstructor
            && $codeIssue->property_id === 'Symfony\Contracts\Service\ServiceSubscriberTrait::$container'
        ) {
            return false;
        }

        return null;
    }
}
```
